# http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/sys-libs/timezone-data/files/2013f-makefile.patch
# added some fixes for the file-layout (jue)
#
# - Support env CC
# - Don't build/install libtz.a
# - Don't install man-pages provided by man-pages
# - Move zic, zdump to sbin and tzselect to bin ala glibc
# - Install posix zoneinfo into zoneinfo/posix/ instead of zoneinfo-posix/ ala glibc
# - Install leaps zoneinfo into zoneinfo/right/ ala glibc


diff -Nru tzcode.orig/Makefile tzcode/Makefile
--- tzcode.orig/Makefile	2015-08-12 11:00:16.696813835 +0200
+++ tzcode/Makefile	2015-08-12 11:00:45.296792265 +0200
@@ -44,21 +44,21 @@
 
 # Everything gets put in subdirectories of. . .
 
-TOPDIR=		/usr/local
+TOPDIR=		/usr
 
 # "Compiled" time zone information is placed in the "TZDIR" directory
 # (and subdirectories).
 # Use an absolute path name for TZDIR unless you're just testing the software.
 
 TZDIR_BASENAME=	zoneinfo
-TZDIR=		$(TOPDIR)/etc/$(TZDIR_BASENAME)
+TZDIR=		$(TOPDIR)/share/$(TZDIR_BASENAME)
 
 # Types to try, as an alternative to time_t.  int64_t should be first.
 TIME_T_ALTERNATIVES= int64_t int32_t uint32_t uint64_t
 
 # The "tzselect", "zic", and "zdump" commands get installed in. . .
 
-ETCDIR=		$(TOPDIR)/etc
+SBINDIR=		$(TOPDIR)/sbin
 
 # If you "make INSTALL", the "date" command gets installed in. . .
 
@@ -320,8 +320,10 @@
 
 #MAKE=		make
 
-cc=		cc
-CC=		$(cc) -DTZDIR=\"$(TZDIR)\"
+CC+=		-DTZDIR=\"$(TZDIR)\"
+ifeq ($(NLS),1)
+CC+= -DHAVE_GETTEXT=1 -DTZ_DOMAIN=\"libc\"
+endif
 
 AR=		ar
 
@@ -371,19 +373,18 @@
 ALL:		all date $(ENCHILADA)
 
 install:	all $(DATA) $(REDO) $(MANS)
-		mkdir -p $(DESTDIR)$(ETCDIR) $(DESTDIR)$(TZDIR) \
-			$(DESTDIR)$(LIBDIR) \
+		mkdir -p $(DESTDIR)$(SBINDIR) $(DESTDIR)$(TZDIR) \
+			$(DESTDIR)$(LIBDIR) $(DESTDIR)$(BINDIR) \
 			$(DESTDIR)$(MANDIR)/man3 $(DESTDIR)$(MANDIR)/man5 \
 			$(DESTDIR)$(MANDIR)/man8
 		$(ZIC) -y $(YEARISTYPE) \
 			-d $(DESTDIR)$(TZDIR) -l $(LOCALTIME) -p $(POSIXRULES)
 		cp -f iso3166.tab $(ZONETABLES) $(DESTDIR)$(TZDIR)/.
-		cp tzselect zic zdump $(DESTDIR)$(ETCDIR)/.
+		cp zic zdump $(DESTDIR)$(SBINDIR)/.
+		cp tzselect $(DESTDIR)$(BINDIR)/.
 		cp libtz.a $(DESTDIR)$(LIBDIR)/.
 		$(RANLIB) $(DESTDIR)$(LIBDIR)/libtz.a
 		cp -f newctime.3 newtzset.3 $(DESTDIR)$(MANDIR)/man3/.
-		cp -f tzfile.5 $(DESTDIR)$(MANDIR)/man5/.
-		cp -f tzselect.8 zdump.8 zic.8 $(DESTDIR)$(MANDIR)/man8/.
 
 INSTALL:	ALL install date.1
 		mkdir -p $(DESTDIR)$(BINDIR) $(DESTDIR)$(MANDIR)/man1
@@ -426,19 +427,17 @@
 # You must replace all of $(TZDIR) to switch from not using leap seconds
 # to using them, or vice versa.
 right_posix:	right_only leapseconds
-		rm -fr $(DESTDIR)$(TZDIR)-leaps
-		ln -s $(TZDIR_BASENAME) $(DESTDIR)$(TZDIR)-leaps || \
-		  $(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)-leaps \
+		rm -fr $(DESTDIR)$(TZDIR)/right
+		  $(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)/right \
 			-L leapseconds $(TDATA)
-		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)-posix \
+		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)/posix \
 			-L /dev/null $(TDATA)
 
 posix_right:	posix_only leapseconds
 		rm -fr $(DESTDIR)$(TZDIR)-posix
-		ln -s $(TZDIR_BASENAME) $(DESTDIR)$(TZDIR)-posix || \
-		  $(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)-posix \
+		  $(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)/posix \
 			-L /dev/null $(TDATA)
-		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)-leaps \
+		$(ZIC) -y $(YEARISTYPE) -d $(DESTDIR)$(TZDIR)/right \
 			-L leapseconds $(TDATA)
 
 posix_packrat:	posix_only backzone
@@ -467,7 +466,7 @@
 		chmod +x $@
 
 check:		check_character_set check_white_space check_links check_sorted \
-		  check_tables check_web
+		  check_tables
 
 check_character_set: $(ENCHILADA)
 		LC_ALL=en_US.utf8 && export LC_ALL && \
